packages/harpguru-core/src/utils/reduce-layoutfacts-to-dynamicsizes/reduce-layoutfacts-to-dynamicsizes.ts

```diff
@@ -134,38 +127,29 @@ export const useSizes = (): SizeSchemes => {
     labelIconSize: dynamicSeedSize * labelIconSize,
   } as const
 
-  const staticEquivalentColumnCount = 12
-  const staticSeedSize =
-    longEdge /
-    (columnWidth * staticEquivalentColumnCount +
-      fragmentGutter * includingHarpFaceEdgesGutterCount +
-      labelProtrusion +
-      legendWidth)
-
-  const staticSizes: SizeScheme = {
-    0: staticSeedSize * relativeSizes[0],
-    1: staticSeedSize * relativeSizes[1],
-    2: staticSeedSize * relativeSizes[2],
-    3: staticSeedSize * relativeSizes[3],
-    4: staticSeedSize * relativeSizes[4],
-    5: staticSeedSize * relativeSizes[5],
-    6: staticSeedSize * relativeSizes[6],
-    7: staticSeedSize * relativeSizes[7],
-    8: staticSeedSize * relativeSizes[8],
-    9: staticSeedSize * relativeSizes[9],
-    10: staticSeedSize * relativeSizes[10],
-    11: staticSeedSize * relativeSizes[11],
-    columnWidth: staticSeedSize * columnWidth,
-    rowHeight: staticSeedSize * rowHeight,
-    legendWidth: staticSeedSize * legendWidth,
-    zoomSlideWidth: staticSeedSize * zoomSlideWidth,
-    fragmentGutter: staticSeedSize * fragmentGutter,
-    labelProtrusion: staticSeedSize * labelProtrusion,
-    labelIconSize: staticSeedSize * labelIconSize,
-  } as const
+  if (prevSizes && isMatch(prevSizes, dynamicSizes)) return prevSizes
+  return dynamicSizes
+}
 
-  return {
-    dynamicSizes,
-    staticSizes,
-  }
+const isMatch = (prevSizes: SizeScheme, nextSizes: SizeScheme): boolean => {
```

Comment:
Is there a way to just iterate over every member of the scheme?

That would be the safest approach but I've had typing issues while using `Object.keys`.

Reply:
There are ways. I don't know if I like what I'm reading at the moment. There's nothing which seems typesafe anyway. Raised as techdebt in https://github.com/js-jslog/harpguru/issues/138#issuecomment-989159471